cpp/src/arrow/tensor-test.cc

```diff
@@ -104,6 +104,48 @@ TEST(TestTensor, ZeroDimensionalTensor) {
   ASSERT_EQ(t.strides().size(), 1);
 }
 
+TEST(TestTensor, CountNonZeroForZeroDimentionalTensor) {
+  std::vector<int64_t> shape = {0};
```

Comment:
This is a one-dimensional tensor, not a zero-dimensional one, right? Or perhaps you meant a zero-sized tensor (which is a different thing).